File: repeated10x-10kRandomData-total100k/BubbleSort/BubbleSort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    FILE *file = fopen("repeated10x_10kRandomData_total100k.txt", "r");
    if (!file) {
        printf("Error: Cannot open repeated10x_10kRandomData_total100k.txt\n");
        return 1;
    }

    int n = 25000;
    int *arr = (int *)malloc(n * sizeof(int));
    if (!arr) {
        printf("Memory allocation failed\n");
        return 1;
    }

    for (int i = 0; i < n; i++) fscanf(file, "%d", &arr[i]);
    fclose(file);

    clock_t start = clock();
    bubbleSort(arr, n);
    clock_t end = clock();

    double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;
    printf("Bubble Sort Execution Time: %.6f seconds\n", time_taken);

    free(arr);
    return 0;
}

File: repeated10x-10kRandomData-total100k/BubbleSort/BubbleSort.py
import time

def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def main():
    try:
        with open("repeated10x_10kRandomData_total100k.txt", "r") as file:
            arr = list(map(int, file.read().split()))
    except FileNotFoundError:
        print("Error: repeated10x_10kRandomData_total100k.txt file not found!")
        return

    start_time = time.time()
    bubble_sort(arr)
    end_time = time.time()

    execution_time = end_time - start_time
    print(f"Bubble Sort Execution Time: {execution_time:.6f} seconds")

if __name__ == "__main__":
    main()
